lib/Optimizer/Partition.cpp

```diff
@@ -96,7 +96,7 @@ NodeFunctionMap selectBasicBlockPartitions(Function *F) {
 
     // If node has only one input, and that input has only one output, place it
     // in the same partition.
-    auto *in = singleNonVariableInput(node);
+    auto *in = singleNonStorageInput(node);
```

Comment:
Just to be sure -- was the intention here for any Storage node? Or was it really intended more for what are now Placeholder inputs? @bertmaher